### PR TITLE
Add filterable entities method

### PR DESCRIPTION
**Pull request by themadcreator**

### CODE REVIEW

**Comment by themadcreator** [comment]
Side note: I could also expose the `_getEntityStore` method and the `_lightweightPlotEntityToPlotEntity` method directly.

File: src/plots/plot.ts [hunk 1]
```diff
@@ -785,6 +785,23 @@ export class Plot extends Component {
...
+   *
+   * @param where A filter predicate that is applied to lightweight entities
+   * before upconverting them to full plot entities
+   * @return {Plots.PlotEntity[]}
+   */
+  public filterEntities(where: (entity: Plots.ILightweightPlotEntity) => boolean): Plots.IPlotEntity[] {
+      return this._getEntityStore()
+        .entities()
+        .filter(where)
+        .map((entity) => this._lightweightPlotEntityToPlotEntity(entity));
```
Comment (giladgray) [inline src/plots/plot.ts]: bizarre indentation here. sometimes 2 spaces, sometimes 4.

File: src/plots/plot.ts [hunk 2]
```diff
@@ -785,6 +785,23 @@ export class Plot extends Component {
...
+   * Gets the `Plots.PlotEntity`s in the plot, whereby the
+   * `Plots.ILightweightPlotEntity`s are filtered by the provided predicate.
+   * Since `_lightweightPlotEntityToPlotEntity` can be expensive, this method
+   * can be useful for performance optimization.
+   *
+   * @param where A filter predicate that is applied to lightweight entities
+   * before upconverting them to full plot entities
+   * @return {Plots.PlotEntity[]}
+   */
+  public filterEntities(where: (entity: Plots.ILightweightPlotEntity) => boolean): Plots.IPlotEntity[] {
```
Comment (giladgray) [inline src/plots/plot.ts]: add `index: number` arg?

**Review by giladgray — APPROVED** [review]
fix that whitespace?

**Pull request merged**